### ISSUE TITLE
I push the same image twice, but get the different digest.

### ISSUE THREAD

**Issue by Hyzhou** [docker/distribution#2066]
[root@iZ62ivvw3i1Z ~]# docker push registry.cn-hangzhou.aliyuncs.com/hyzhou/hyzhou:test1
The push refers to a repository [registry.cn-hangzhou.aliyuncs.com/hyzhou/hyzhou]
5f70bf18a086: Layer already exists
899268416f97: Pushed
e88b3f82283b: Pushed
test1: digest: sha256:f746e837828daf1a8a1f1743de5c9a95d3b14112f2245912683226d4545518d1 size: 1146
[root@iZ62ivvw3i1Z ~]# docker push registry.cn-hangzhou.aliyuncs.com/hyzhou/hyzhou:test1
The push refers to a repository [registry.cn-hangzhou.aliyuncs.com/hyzhou/hyzhou]
5f70bf18a086: Layer already exists
899268416f97: Layer already exists
e88b3f82283b: Layer already exists
test1: digest: sha256:91ab276ad332a0b1e93ac090386c9b54df9088679e87e5871faf2b2844e71fc4 size: 1124

I don't know, why the layel size different?

**Comment by Hyzhou** [comment]
I try to download that two blobs. the different is value of the mediaType.

First time to push the image.
```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "config": {
      "mediaType": "application/vnd.docker.container.image.v1+json",
      "size": 2248,
      "digest": "sha256:7c08ec48fc6d43fb5bd42f75fb9441be2587d08acdde81029784c7fc478ad123"
   },
   "layers": [
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 668151,
         "digest": "sha256:56bec22e355981d8ba0878c6c2f23b21f422f30ab0aba188b54f1ffeff59c190"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 205,
         "digest": "sha256:a486a2872031c757c144eb781e46041707d882f47e3e30c10783490e10bc37ee"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 32,
         "digest": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 32,
         "digest": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      }
   ]
}
```

Push the image again.
```
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
   "config": {
      "mediaType": "application/octet-stream",
      "size": 2248,
      "digest": "sha256:7c08ec48fc6d43fb5bd42f75fb9441be2587d08acdde81029784c7fc478ad123"
   },
   "layers": [
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 668151,
         "digest": "sha256:56bec22e355981d8ba0878c6c2f23b21f422f30ab0aba188b54f1ffeff59c190"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 205,
         "digest": "sha256:a486a2872031c757c144eb781e46041707d882f47e3e30c10783490e10bc37ee"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 32,
         "digest": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
         "size": 32,
         "digest": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      }
   ]
}
```

**Comment by zhanghuanzhong** [comment]
I think you have run a container with the first image, right?

2016-11-15 17:03 GMT+08:00 Hyzhou <notifications@github.com>:

**Comment by aaronlehmann** [comment]
This looks like a duplicate of #1621. It should have been fixed in Docker 1.11.1 (see https://github.com/docker/docker/pull/21949). What version are you running?

**Comment by Hyzhou** [comment]
[root@iZ62ivvw3i1Z ~]# docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:23:11 2016
 OS/Arch:      linux/amd64

**Comment by Hyzhou** [comment]
I have not run any container with the first image.
I just pull the image, retag it, and push the image to aliyun registry twice.

**Comment by aaronlehmann** [comment]
Sorry, it actually looks like it was fixed in 1.12.0: https://github.com/docker/docker/issues/24360

Can you try upgrading to 1.12.3?

**Comment by Hyzhou** [comment]
@aaronlehmann 
I have find other docker client. It is already fixed in 1.12.1.
I will try the latest docker version next time before the issues posted.
